docs/DesignDocs/RemoteAttestationCollaterals.md

```diff
@@ -0,0 +1,457 @@
+       +----------------+                     +-----------------+
+
+                           Figure 1: RATS Roles
+
+### Roles
+
+RATS roles are implemented by principals that possess cryptographic
+keys used to protect and authenticate Claims or Results.
+
+#### Attester:
+An Attestation Function that creates Evidence by
+collecting, formatting and protecting (e.g., signing) Claims.  It
+presents Evidence to a Verifier using a conveyance mechanism or
+protocol.
+
+#### Verifier:
+An Attestation Function that accepts Evidence from an Attester using
+a conveyance mechanism or protocol.  It also accepts Known-Good-Values
+and Endorsments from an Asserter using a conveyance mechanism or protocol.
+It verifies the protection mechanisms, parses and appraises Evidence
+according to good-known valid (or known-invalid) Claims and Endorsments.
+It produces Attestation Results that are formatted and protected (e.g.,
+signed).  It presents Attestation Results to a Relying Party using
+a conveyance mechanism or protocol.
+
+#### Asserter:
+An Attestation Function that generates reference Claims
+about both the Attesting Computing Environment and the Attested
+Computing Environment.  The manufacturing and development
+processes are presumed to be trustworthy processes.  In other
+words the Asserter is presumed, by a Verifier, to produce valid
+Claims.  The function collects, formats and protects (e.g. signs)
+valid Claims known as Endorsements and Known-Good-Values.  It
+presents provable Claims to a Verifier using a conveyance
+mechanism or protocol.
+
+#### Relying Party:
+An Attestation Function that accepts Attestation
+Results from a Verifier using a conveyance mechanism or protocol.
+It assesses Attestation Results protections, parses and assesses
+Attestation Results according to an assessent context (Note:
+definition of the assessment context is out-of-scope).
+
+
+Motivation
+----------
+
+Currently, the existing endorsements used for Intel SGX quote verification are not exposed to the user.  This makes it difficult for the verifier to specify his/her own set of policies.  Adding these new APIs allows the verifier to specify a validation policy of his/her choosing.  Possible policies:
+1. The verifier has the option to specify its set of endorsements during verification.
+2. The verifier has the option to provide a datetime to use during verification.  This datetime specifies the date and time at which the verifier wants to do the verification.  If no datetime is provided, the datetime when the endorsements were created is used during verification.  The verifier can provide a datetime in the past, enabling auditing of the evidence and endorsements.
+
+
+User Experience
+---------------
+
+There are 2 scenarios.
+
+### 1. Verifier is provided with endorsements:
+In this scenario the attester/asserter provides the evidence and endorsements to the verifier.  The verifier is then free to use these to verify the TEE.
```

Comment:
>attester/asserter provides the evidence and endorsements to the verifier [](start = 21, length = 72)

Just to make it clear - attester provides the evidence and the asserter provides the endorsement.

Reply:
It could. Either the attester can fetch it from the attester and then send a bundled evidence + endorsement to the verifier, or the verifier could fetch it themselves.